spec="%gcc@8.3.1 build_type=Release" warning-as-error

Saw this warning-as-error in a LiDO merge request when spec="%gcc@8.3.1 build_type=Release"; partial log below:

```console
==> Installing serac-develop-nlbb25r7isuacsdfv3p6gtvexjopqdbe
==> No binary for serac-develop-nlbb25r7isuacsdfv3p6gtvexjopqdbe found: installing from source
==> serac: Executing phase: 'hostconfig'
==> serac: Executing phase: 'cmake'
==> serac: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8'
4 errors found in build log:
     157    cd /usr/WS1/oxberry1/.gitlab-runner/builds/XvEgLX5S/3/gitlab/topopt
            /lido-2.0/uberenv_libs/spack/var/spack/stage/spack-stage-serac-deve
            lop-nlbb25r7isuacsdfv3p6gtvexjopqdbe/spack-build-nlbb25r/src/serac/
            coefficients && /usr/tce/packages/cmake/cmake-3.14.5/bin/cmake -E c
            make_link_script CMakeFiles/serac_coefficients.dir/link.txt --verbo
            se=1
     158    /bin/ar qc ../../../lib/libserac_coefficients.a  CMakeFiles/serac_c
            oefficients.dir/loading_functions.cpp.o CMakeFiles/serac_coefficien
            ts.dir/coefficient_extensions.cpp.o
     159    /bin/ranlib ../../../lib/libserac_coefficients.a
     160    make[2]: Leaving directory `/usr/WS1/oxberry1/.gitlab-runner/builds
            /XvEgLX5S/3/gitlab/topopt/lido-2.0/uberenv_libs/spack/var/spack/sta
            ge/spack-stage-serac-develop-nlbb25r7isuacsdfv3p6gtvexjopqdbe/spack
            -build-nlbb25r'
     161    [ 25%] Built target serac_coefficients
     162    /usr/WS1/oxberry1/.gitlab-runner/builds/XvEgLX5S/3/gitlab/topopt/li
            do-2.0/uberenv_libs/spack/var/spack/stage/spack-stage-serac-develop
            -nlbb25r7isuacsdfv3p6gtvexjopqdbe/spack-src/src/serac/infrastructur
            e/input.cpp: In member function 'std::unique_ptr<mfem::Coefficient>
             serac::input::CoefficientInputOptions::constructScalar() const':
  >> 163    /usr/WS1/oxberry1/.gitlab-runner/builds/XvEgLX5S/3/gitlab/topopt/li
            do-2.0/uberenv_libs/spack/var/spack/stage/spack-stage-serac-develop
            -nlbb25r7isuacsdfv3p6gtvexjopqdbe/spack-src/src/serac/infrastructur
            e/input.cpp:154:46: error: potential null pointer dereference [-Wer
            ror=null-dereference]
     164         mfem::Vector pw_constants(max_attr_elem->first);
     165                                   ~~~~~~~~~~~~~~~^~~~~
     166    cc1plus: all warnings being treated as errors
  >> 167    make[2]: *** [src/serac/infrastructure/CMakeFiles/serac_infrastruct
            ure.dir/input.cpp.o] Error 1
     168    make[2]: Leaving directory `/usr/WS1/oxberry1/.gitlab-runner/builds
            /XvEgLX5S/3/gitlab/topopt/lido-2.0/uberenv_libs/spack/var/spack/sta
            ge/spack-stage-serac-develop-nlbb25r7isuacsdfv3p6gtvexjopqdbe/spack
            -build-nlbb25r'
  >> 169    make[1]: *** [src/serac/infrastructure/CMakeFiles/serac_infrastruct
            ure.dir/all] Error 2
     170    make[1]: Leaving directory `/usr/WS1/oxberry1/.gitlab-runner/builds
            /XvEgLX5S/3/gitlab/topopt/lido-2.0/uberenv_libs/spack/var/spack/sta
            ge/spack-stage-serac-develop-nlbb25r7isuacsdfv3p6gtvexjopqdbe/spack
            -build-nlbb25r'
  >> 171    make: *** [all] Error 2
See build log for details:
  /usr/WS1/oxberry1/.gitlab-runner/builds/XvEgLX5S/3/gitlab/topopt/lido-2.0/uberenv_libs/spack/var/spack/stage/spack-stage-serac-develop-nlbb25r7isuacsdfv3p6gtvexjopqdbe/spack-build-out.txt
```

--- a/src/serac/infrastructure/input.cpp
+++ b/src/serac/infrastructure/input.cpp
@@ -160,9 +160,7 @@ std::unique_ptr<mfem::Coefficient> CoefficientInputOptions::constructScalar() co
     auto max_attr_elem = std::max_element(scalar_pw_const.begin(), scalar_pw_const.end(),
                                           [](auto a, auto b) { return a.first < b.first; });
 
-    // NOTE(oxberry1@llnl.gov): gcc 8.3.1 Release builds emit
-    // "potential null pointer dereference [-Werror=null-dereference]"
-    // for max_attr_elem->first if not guarded by an if statement
+    // Check for a valid iterator to avoid compiler warnings
     if (max_attr_elem != scalar_pw_const.end()) {
       // Create an mfem vector for the attributes
       // Note that this vector expects zero indexing
@@ -175,11 +173,11 @@ std::unique_ptr<mfem::Coefficient> CoefficientInputOptions::constructScalar() co
 
       // Create the MFEM coefficient
       return std::make_unique<mfem::PWConstCoefficient>(pw_constants);
-    } // Control falls through to error if max_attr_elem is scalar_pw_const.end()
-  } // Control falls through to error if none of the if-else conditions is true
-  
+    }
+  }
+
   SLIC_ERROR_ROOT(
-        "Trying to build a scalar coefficient without specifying a scalar_function, constant, or piecewise_constant.");
+      "Trying to build a scalar coefficient without specifying a scalar_function, constant, or piecewise_constant.");
   return nullptr;
 }
 

--- a/src/serac/infrastructure/input.cpp
+++ b/src/serac/infrastructure/input.cpp
@@ -149,23 +149,27 @@ std::unique_ptr<mfem::Coefficient> CoefficientInputOptions::constructScalar() co
     auto max_attr_elem = std::max_element(scalar_pw_const.begin(), scalar_pw_const.end(),
                                           [](auto a, auto b) { return a.first < b.first; });
 
-    // Create an mfem vector for the attributes
-    // Note that this vector expects zero indexing
-    mfem::Vector pw_constants(max_attr_elem->first);
-    pw_constants = 0.0;
+    // NOTE(oxberry1@llnl.gov): gcc 8.3.1 Release builds emit
+    // "potential null pointer dereference [-Werror=null-dereference]"
+    // for max_attr_elem->first if not guarded by an if statement
+    if (max_attr_elem != scalar_pw_const.end()) {
+      // Create an mfem vector for the attributes
+      // Note that this vector expects zero indexing
+      mfem::Vector pw_constants(max_attr_elem->first);
+      pw_constants = 0.0;
 
-    for (auto& entry : scalar_pw_const) {
-      pw_constants(entry.first - 1) = entry.second;
-    }
+      for (auto& entry : scalar_pw_const) {
+        pw_constants(entry.first - 1) = entry.second;
+      }
 
-    // Create the MFEM coefficient
-    return std::make_unique<mfem::PWConstCoefficient>(pw_constants);
-
-  } else {
-    SLIC_ERROR_ROOT(
+      // Create the MFEM coefficient
+      return std::make_unique<mfem::PWConstCoefficient>(pw_constants);
+    } // Control falls through to error if max_attr_elem is scalar_pw_const.end()
+  } // Control falls through to error if none of the if-else conditions is true
+  
+  SLIC_ERROR_ROOT(
         "Trying to build a scalar coefficient without specifying a scalar_function, constant, or piecewise_constant.");
-    return nullptr;
-  }
+  return nullptr;
 }
 
 void CoefficientInputOptions::defineInputFileSchema(axom::inlet::Container& container)

